internal/http/errors_test.go

```diff
@@ -0,0 +1,96 @@
+package http
+
+import (
+	"errors"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestWriteErrorJSON(t *testing.T) {
+	for _, test := range []struct {
+		name         string
+		writeFn      func() *httptest.ResponseRecorder
```

Comment:
This is unnecessarily complex. Make this `err error` instead and create the response recorder in the body of the for loop.